api_app/connectors_manager/connectors/opencti.py

```diff
@@ -0,0 +1,215 @@
+            else:
+                obs_type = INTELOWL_OPENCTI_TYPE_MAP["generic"]  # text
+        else:
+            obs_type = INTELOWL_OPENCTI_TYPE_MAP[self._job.observable_classification]
+
+        return obs_type
+
+    def generate_observable_data(self) -> dict:
+        observable_data = {"type": self.get_observable_type()}
+        if self._job.is_sample:
+            observable_data["name"] = self._job.file_name
+            observable_data["hashes"] = helpers.generate_hashes(self.job_id)
+        elif (
+            self._job.observable_classification == "hash"
+            and observable_data["type"] == "file"
+        ):
+            # add hash instead of value
+            matched_type = helpers.get_hash_type(self._job.observable_name)
+            observable_data["hashes"] = {matched_type: self._job.observable_name}
+        else:
+            observable_data["value"] = self._job.observable_name
+
+        return observable_data
+
+    @property
+    def organization_id(self) -> str:
+        # check if presaved
+        if self.opencti_db.get("organization_id", None) is None:
+            # Create author (if not exists)
+            org = pycti.Identity(self.opencti_instance).create(
+                type="Organization",
+                name="IntelOwl",
+                description=(
+                    "Intel Owl is an Open Source Intelligence, or OSINT solution"
+                    " to get threat intelligence data about a specific file, an IP or a"
+                    " domain from a single API at scale. [Visit the project on GitHub]"
+                    "(https://github.com/intelowlproject/IntelOwl/)"
+                ),
+            )
+            self.opencti_db["organization_id"] = org["id"]
+        return self.opencti_db["organization_id"]
+
+    @property
+    def marking_definition_id(self) -> str:
+        # check if not presaved
+        if self.opencti_db.get("marking_definition_id", None) is None:
+            # Create the marking definition (if not exists)
+            md = pycti.MarkingDefinition(self.opencti_instance).create(
+                definition_type="TLP",
+                definition=f"TLP:{self.tlp['type'].upper()}",
+                x_opencti_color=self.tlp["color"].upper(),
+                x_opencti_order=self.tlp["x_opencti_order"],
+            )
+            self.opencti_db["marking_definition_id"] = md["id"]
+        return self.opencti_db["marking_definition_id"]
+
+    def run(self):
+        # set up client
+        self.opencti_instance = pycti.OpenCTIApiClient(
+            url=self.__url_name,
+            token=self.__api_key,
+            ssl_verify=self.ssl_verify,
+            proxies=self.proxies,
+        )
+
+        self.opencti_db = {}
+        # get presaved values stored in MEDIA_ROOT
+        if os.path.exists(OPENCTI_DB):
+            with open(OPENCTI_DB, "r") as file:
+                self.opencti_db = json.load(file)
```

Comment:
I do not like this. If we want to persist, we can use django's cache framework. The [django-cache-memoize](https://github.com/peterbe/django-cache-memoize) library is already installed,

Reply:
or simply use the `django.core.cache.cache.set(cache_key, function_out, expiration)`

Reply:
Thanks, I'll use `cache_memoize` here.